ovn_cluster.sh

```diff
@@ -526,32 +629,42 @@ function start() {
-            ${RUNC_CMD} exec ${CENTRAL_NAME} ${OVNCTL_PATH} --ovn-manage-ovsdb=no start_northd
-
-        elif [ "$OVN_DB_CLUSTER" = "yes" ]; then
-            start-db-cluster
-            central=${CENTRAL_NAME}-1
-        else
-            ${RUNC_CMD} exec ${CENTRAL_NAME} ${OVNCTL_PATH} start_northd
-            sleep 2
-        fi
+        for name in "${CENTRAL_NAMES[@]}"; do
+            central=${name}
+            if [ "$ENABLE_ETCD" == "yes" ]; then
+                echo "Starting ovsdb-etcd in ${name} container"
+                ${RUNC_CMD} exec --detach ${name} bash -c "/run_ovsdb_etcd.sh"
+                sleep 2
+                ${RUNC_CMD} exec --detach ${name} bash -c "/run_ovsdb_etcd_sb.sh"
+                ${RUNC_CMD} exec --detach ${name} bash -c "/run_ovsdb_etcd_nb.sh"
+                ${RUNC_CMD} exec ${name} ${OVNCTL_PATH} --ovn-manage-ovsdb=no start_northd
+
+            elif [ "$OVN_DB_CLUSTER" = "yes" ]; then
+                start-db-cluster ${name}
+                central=${name}-1
+            else
+                ${RUNC_CMD} exec ${name} ${OVNCTL_PATH} start_northd
+                sleep 2
+            fi
 
-        if [ "$ENABLE_SSL" == "yes" ]; then
-            ${RUNC_CMD} exec ${central} ovn-nbctl set-ssl ${SSL_CERTS_PATH}/ovn-privkey.pem  ${SSL_CERTS_PATH}/ovn-cert.pem ${SSL_CERTS_PATH}/pki/switchca/cacert.pem
-            ${RUNC_CMD} exec ${central} ovn-sbctl set-ssl ${SSL_CERTS_PATH}/ovn-privkey.pem  ${SSL_CERTS_PATH}/ovn-cert.pem ${SSL_CERTS_PATH}/pki/switchca/cacert.pem
-        fi
-        ${RUNC_CMD} exec ${central} ovn-nbctl set-connection p${REMOTE_PROT}:6641
-        ${RUNC_CMD} exec ${central} ovn-nbctl set connection . inactivity_probe=180000
+            if [ "$ENABLE_SSL" == "yes" ]; then
+                ${RUNC_CMD} exec ${central} ovn-nbctl set-ssl ${SSL_CERTS_PATH}/ovn-privkey.pem  ${SSL_CERTS_PATH}/ovn-cert.pem ${SSL_CERTS_PATH}/pki/switchca/cacert.pem
+                ${RUNC_CMD} exec ${central} ovn-sbctl set-ssl ${SSL_CERTS_PATH}/ovn-privkey.pem  ${SSL_CERTS_PATH}/ovn-cert.pem ${SSL_CERTS_PATH}/pki/switchca/cacert.pem
+            fi
+            ${RUNC_CMD} exec ${central} ovn-nbctl set-connection p${REMOTE_PROT}:6641
+            ${RUNC_CMD} exec ${central} ovn-nbctl set connection . inactivity_probe=180000
+
+            ${RUNC_CMD} exec ${central} ovn-nbctl set NB_Global . name=${central} \
+                options:ic-route-adv=true options:ic-route-learn=true
 
-        ${RUNC_CMD} exec ${central} ovn-sbctl set-connection p${REMOTE_PROT}:6642
-        ${RUNC_CMD} exec ${central} ovn-sbctl set connection . inactivity_probe=180000
+            ${RUNC_CMD} exec ${central} ovn-sbctl set-connection p${REMOTE_PROT}:6642
+            ${RUNC_CMD} exec ${central} ovn-sbctl set connection . inactivity_probe=180000
+        done
+
+        # start ovn-ic dbs
+        if [ $CENTRAL_COUNT -gt 1 ]; then
+            start-ovn-ic
+        fi
```

Comment:
With the new environment variable OVN_START_IC_DBS we could just always call this no need to check on $CENTRAL_COUNT > 1.

Reply:
@dceara ack, I will fix it